### PR TITLE
Add tests for list

### LINKED ISSUE #147: Add tests for List Command
* to test the random filter of n flashcards (with a stub and seed)
* additional tests that might have been missed out

refer to #108

### PR DESCRIPTION
**Pull request by kwokieee**
### Description

- Created tests for listing a random number of flashcards.
- Added equals test for `ListCommand`.
- Fixed some errors spotted in Javadocs for `FlashcardInGivenFlashcardListPredicate`.

Closes #147.

### Type of change

- [x] Adding tests

### Checklist

- [x] I have written testcases to cover all new methods
- [x] I have built and manually tested this code
- [ ] I have updated the User Guide
- [ ] I have updated the Developer Guide
- [ ] I have updated my individual project portfolio
- [ ] I have included a preview deployment link to the updated project documentation

### CODE REVIEW

**Review by jhlee1997 — APPROVED** [review]
LGTM

File: src/test/java/lingogo/logic/commands/ListCommandTest.java [hunk 1]
```diff
@@ -46,4 +66,9 @@ public void execute_slideshowActive_throwsCommandException() {
 
         assertCommandFailure(new ListCommand(), model, expectedMessage);
     }
+
+    @Test
+    public void equals() {
+
+    }
```
Comment (chownces) [inline src/test/java/lingogo/logic/commands/ListCommandTest.java:73]: Missing method body?

File: src/main/java/lingogo/model/flashcard/FlashcardInGivenFlashcardListPredicate.java [hunk 2]
```diff
@@ -23,6 +27,19 @@ public FlashcardInGivenFlashcardListPredicate(List<Flashcard> givenFlashcardList
...
+     * is in the list of specified flashcards.
+     * @param flashcards The list of specified flashcards.
+     */
+    public FlashcardInGivenFlashcardListPredicate(Flashcard... flashcards) {
+        HashSet<Flashcard> tempSet = new HashSet<>();
+        for (Flashcard flashcard : flashcards) {
+            tempSet.add(flashcard);
+        }
+        this.givenFlashcardSet = tempSet;
+    }
```
Comment (chownces) [inline src/main/java/lingogo/model/flashcard/FlashcardInGivenFlashcardListPredicate.java:41]: We can just use the constructor above? (just create a `new List(phrase1, phrase2, ...)` inside the test)

File: src/test/java/lingogo/logic/commands/ListCommandTest.java [hunk 3]
```diff
@@ -37,6 +41,22 @@ public void execute_listWithInvalidIndex_throwsCommandException() {
...
+        int n = 3;
+
+        // With seed = 10, flashcards expected to be selected are:
+        // AFTERNOON_CHINESE_FLASHCARD, NIGHT_CHINESE_FLASHCARD and SORRY_CHINESE_FLASHCARD
+        expectedModel.updateFilteredFlashcardList(new FlashcardInGivenFlashcardListPredicate(
+                AFTERNOON_CHINESE_FLASHCARD, NIGHT_CHINESE_FLASHCARD, SORRY_CHINESE_FLASHCARD));
+        String expectedMessage = String.format(ListCommand.MESSAGE_SUCCESS_SHUFFLED, n);
+
+        assertCommandSuccess(new ListCommand(n, seed), model, expectedMessage, expectedModel);
     }
```
Comment (chownces) [inline src/test/java/lingogo/logic/commands/ListCommandTest.java:60]: would be nice to add another positive test case with a different seed too!

**Review by chownces — CHANGES_REQUESTED** [review]

**Review by chownces — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**